backend/src/app.js

```diff
@@ -0,0 +1,32 @@
+const fastify = require('fastify')({ logger: true });
+
+const { pgBossQueue } = require('./pg');
+
+fastify.route({
```

Comment:
Can you elaborate on how you were planning to trigger this job? I had envisioned this working w/o any kind of web server to trigger jobs. For example, using a "cron job" (but the Heroku equivalent) to queue a job every N minutes.

Reply:
Netlify hook should call this endpoint. Even cron job (which will kick off for periodic batch re-scraping of all organisations' data, for example), will need to call some simplistic web endpoint which just puts a job into a queue.

I don't see how more *direct* it could be. Push jobs directly to PostgreSQL via a query? [graphile/worker supports this](https://github.com/graphile/worker#more-detail-on-scheduling-jobs-through-sql), but not pg-boss (at least, it doesn't document this), so it's a slippery slope. But I also don't see this as a problem.

Even with direct SQL query to create a job, you need some basic auth/protection, I suppose, eventually, so you can't really avoid having a thin web layer.

Cron batch processing in Heroku can also be done by starting a dyno, which will do the batch processing and finish. But this won't work for first-time scraping of newly added orgs, there should be a `worker` listening in the background. And since we need this type of worker anyway, it can handle batch scraping, too.

Reply:
> will need to call some simplistic web endpoint which just puts a job into a queue.

The way we usually do this is to run a script within the project that has the job of simply queueing the jobs you want to run. To elaborate, the script would work like this:

`$ npm run queue-jobs`

* Fetch organizations from Airtable that either A) have not be scraped or B) are "outdated"
* For each organization
  * Queue a job to process that organization

The script runs in a couple seconds and is only responsible for adding the right jobs to the queue. The worker does the rest. This avoids needing the web interface, keeps the logic of which orgs to scrape internal to the app, and removes the dependency on Zapier.